DatadogFlags/Sources/Client/FlagAssignmentsRequest.swift

```diff
@@ -0,0 +1,100 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+extension URLRequest {
+    internal static func flagAssignmentsRequest(
+        url: URL,
+        evaluationContext: FlagsEvaluationContext,
+        context: DatadogContext,
+        customHeaders: [String: String]?
+    ) -> URLRequest {
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+
+        request.setValue("application/vnd.api+json", forHTTPHeaderField: "Content-Type")
+        request.setValue(context.clientToken, forHTTPHeaderField: "dd-client-token")
+
+        if let applicationId = context.additionalContext(ofType: RUMCoreContext.self)?.applicationID {
+            request.setValue(applicationId, forHTTPHeaderField: "dd-application-id")
+        }
```

Comment:
nit/ We might consider reusing [`URLRequestBuilder`](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/Upload/URLRequestBuilder.swift) as the canonical way of constructing requests (though it would require small adjustments). This is definitely a nit - benefits of this builder may be unclear in this context since this intake doesn’t use compression (?) and most headers here aren’t standard.

Reply:
Agree 👍